rules/terraform_required_version.go

```diff
@@ -78,13 +80,53 @@ func (r *TerraformRequiredVersionRule) Check(runner tflint.Runner) error {
 		exists = exists || ok
 	}
 
-	if !exists {
-		return runner.EmitIssue(
-			r,
-			`terraform "required_version" attribute is required`,
-			hcl.Range{},
-		)
+	if exists {
+		return nil
+	}
+
+	if len(body.Blocks) > 0 {
+		return r.emitIssue(body.Blocks[0].DefRange, runner)
 	}
 
-	return nil
+	// If there are no "terraform" blocks, create a hcl.Range from the files
+	var file string
+	for k := range files {
+		file = k
+		break
+	}
+
+	// If there is only one file, use that
+	if len(files) == 1 {
+		return r.emitIssue(hcl.Range{
+			Filename: file,
+			Start:    hcl.InitialPos,
+		}, runner)
+	}
+
+	moduleDirectory := filepath.Dir(file)
+
+	// If there are multiple files, look for terraform.tf or main.tf (in that order)
+	for _, basename := range []string{"terraform.tf", "main.tf"} {
```

Comment:
The result of this may be non-deterministic because the order of the loops is undefined.
https://go.dev/doc/go1#iteration

Reply:
This is a slice, not a map, so the order will be deterministic.

Reply:
Oops, that's my mistake. You're right.